### PR TITLE
Add pixi cloud function

### PR DESCRIPTION
**Pull request by sebastianbenz**
This is the first step in adding the new puppeteer based pixi checks to
amp.dev. This PR adds:

* a google cloud function integrating the new
toolbox-page-experience module
* deploy and bootstrap tasks
* basic api integration test

Subsequent PRs will integrate this into the pix API endpoint. This
version is currently deployed here: https://us-central1-amp-dev-230314.cloudfunctions.net/checkPageExperience?url=https://axios.com

### CODE REVIEW

File: pixi/cloud-function/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  if (!url) {
+    res.status(400).send('Missing url parameter');
+    return;
+  }
+  try {
+    new URL(url);
+  } catch (e) {
+    res.status(400).send(`invalid URL "${url}"`);
+    return;
+  }
```
Comment (matthiasrohmer) [inline pixi/cloud-function/index.js:39]: This method of URL validation has a shortcoming which made us fall back to a RegEx based solution for the Pixi frontend - I can't remember the reason though...

File: platform/lib/utils/project.js [hunk 2]
```diff
@@ -68,6 +68,7 @@ const paths = {
   RECENT_GUIDES_DEST: 'pages/shared/data/recent-guides.yaml',
   BUILD_INFO_PATH: absolute('platform/config/build-info.yaml'),
   THUMBOR_ROOT: absolute('thumbor'),
+  PIXI_ROOT: absolute('pixi/cloud-function'),
```
Comment (matthiasrohmer) [inline platform/lib/utils/project.js:71]: `PIXI_CLOUD_ROOT` maybe? I consider `pixi/` as Pixi's root.

File: gulpfile.js/build.js [hunk 3]
```diff
@@ -167,6 +168,15 @@ async function buildPixi() {
   return sh('npm run build:pixi');
 }
 
+/**
+ * Builds the pixi cloud functions project
+ */
+function buildPixiFunctions() {
+  return sh('npm install', {
+    workingDir: PIXI_ROOT,
+  });
```
Comment (matthiasrohmer) [inline gulpfile.js/build.js:177]: That's just a stub, right?

**Review by matthiasrohmer — APPROVED** [review]
Only nits 🙂

File: gulpfile.js/build.js (hunk 3 above)
Comment (sebastianbenz) [reply gulpfile.js/build.js:177]: how do you mean?  It needs to be its own npm module.

File: pixi/cloud-function/index.js (hunk 1 above)
Comment (sebastianbenz) [reply pixi/cloud-function/index.js:39]: It doesn't check the protocol maybe?

File: platform/lib/utils/project.js (hunk 2 above)
Comment (sebastianbenz) [reply platform/lib/utils/project.js:71]: good suggestion! done

File: pixi/cloud-function/index.js (hunk 1 above)
Comment (sebastianbenz) [reply pixi/cloud-function/index.js:39]: seems like it, using the same approach now

**Pull request merged**